pkg/operator/ceph/object/controller.go

```diff
@@ -292,6 +320,68 @@ func (r *ReconcileCephObjectStore) reconcileCreateObjectStore(cephObjectStore *c
 	return reconcile.Result{}, nil
 }
 
+func (r *ReconcileCephObjectStore) reconcileObjectStoreZone(cephObjectStore *cephv1.CephObjectStore) (reconcile.Result, error) {
+	err := getObjectStoreZone(r.context.RookClientset.CephV1(), cephObjectStore.Namespace, cephObjectStore.Spec.Multisite.Zone)
+	if err != nil {
+		return WaitForRequeueIfObjectStoreNotReady, err
+	}
+
+	logger.Debugf("CephObjectStoreZone resource %s found", cephObjectStore.Spec.Multisite.Zone)
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileCephObjectStore) reconcileCephZone(cephObjectStore *cephv1.CephObjectStore) (reconcile.Result, error) {
+	zoneName := cephObjectStore.Spec.Multisite.Zone
+
+	// get zone group and realm info for getCephZone
+	zone, err := r.context.RookClientset.CephV1().CephObjectStoreZones(cephObjectStore.Namespace).Get(zoneName, metav1.GetOptions{})
+	if err != nil {
+		return WaitForRequeueIfObjectStoreNotReady, err
+	}
+
+	zonegroup, err := r.context.RookClientset.CephV1().CephObjectStoreZoneGroups(cephObjectStore.Namespace).Get(zone.Spec.ZoneGroup, metav1.GetOptions{})
+	if err != nil {
+		return WaitForRequeueIfObjectStoreNotReady, err
+	}
+
+	realm, err := r.context.RookClientset.CephV1().CephObjectStoreRealms(cephObjectStore.Namespace).Get(zonegroup.Spec.Realm, metav1.GetOptions{})
+	if err != nil {
+		return WaitForRequeueIfObjectStoreNotReady, err
+	}
+
+	err = getCephZone(r.context, zoneName, cephObjectStore.Namespace, zonegroup.Name, realm.Name)
+	if err != nil {
+		return WaitForRequeueIfObjectStoreNotReady, err
+	}
+
+	logger.Debugf("Zone %s found in Ceph cluster", zoneName)
+	return reconcile.Result{}, nil
+}
+
+func (r *ReconcileCephObjectStore) updateMultisiteNames(cephObjectStore *cephv1.CephObjectStore) (string, string, string, reconcile.Result, error) {
+	if isMultisite(cephObjectStore.Spec) {
+		zoneName := cephObjectStore.Spec.Multisite.Zone
+		zone, err := r.context.RookClientset.CephV1().CephObjectStoreZones(cephObjectStore.Namespace).Get(zoneName, metav1.GetOptions{})
+		if err != nil {
+			return "", "", "", WaitForRequeueIfObjectStoreNotReady, err
+		}
+
+		zonegroup, err := r.context.RookClientset.CephV1().CephObjectStoreZoneGroups(cephObjectStore.Namespace).Get(zone.Spec.ZoneGroup, metav1.GetOptions{})
+		if err != nil {
+			return "", "", "", WaitForRequeueIfObjectStoreNotReady, err
+		}
+
+		realm, err := r.context.RookClientset.CephV1().CephObjectStoreRealms(cephObjectStore.Namespace).Get(zonegroup.Spec.Realm, metav1.GetOptions{})
+		if err != nil {
+			return "", "", "", WaitForRequeueIfObjectStoreNotReady, err
+		}
+
+		return realm.Name, zonegroup.Name, zone.Name, reconcile.Result{}, nil
```

Comment:
I think only the names are necessary. The only CR that the object store (and generally all the multisite CRs) should be concerned with is the one that's the level above them. In this case it's the zone. The only reason we need the names is so that we can pass --rgw-{realm/zonegroup/zone}={realm/zonegroup/zone}-name to the radosgw-admin commands in the object store. I can't see any reason why we'd need other information from those higher level CRs anytime soon.

Reply:
This makes sense now since the new controllers apply the settings for the realms/zonegroups/zones. This question was based on my false assumption that the object store was doing more.